angular-app/src/app/project-detail/project-detail.component.ts

```diff
@@ -1,61 +1,66 @@
+import { Component, Input, OnInit, OnChanges, SimpleChanges } from '@angular/core';
 import { MatDialog, MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
-import { Router, ActivatedRoute, ParamMap } from '@angular/router'; 
 import { Observable } from 'rxjs';
-import { switchMap } from 'rxjs/operators';
+import { first } from 'rxjs/operators';
 
 import { Project } from '../project';
-import { PROJECTS } from '../mock-projects';
-import { ToggleChatService } from '../toggle-chat.service';
+import { ProjectFile } from '../project-file';
+import { ToggleRightDrawerService } from '../toggle-right-drawer.service';
 import { ProjectService } from '../project.service';
 
 @Component({
   selector: 'app-project-detail',
   templateUrl: './project-detail.component.html',
   styleUrls: ['./project-detail.component.css']
 })
-export class ProjectDetailComponent implements OnInit {
+export class ProjectDetailComponent implements OnInit, OnChanges {
 
-  private files: string[] = ['audio.wav', 'audio.mp3', 'lyric1.txt', 'lyric2.docx']
-  project$: Observable<Project>; // $ postfix simply indicates an Observable and has no function
+  private blobstoreUploadUrl: string;
+  @Input() project$: Observable<Project>;
+  files$: Observable<ProjectFile[]>;
 
   constructor(
-    private route: ActivatedRoute,
-    private router: Router,
     private projectService: ProjectService,
-    private chatService: ToggleChatService,
+    private toggleRightDrawerService: ToggleRightDrawerService,
     public dialog: MatDialog
   ) { }
 
   ngOnInit(): void {
-    console.log("opening project detail view")
-    this.project$ = this.route.paramMap.pipe(
-      switchMap((params: ParamMap) =>
-        this.projectService.getProject(params.get('id')))
-    );
+    this.setBlobstoreUploadUrl();
   }
+  
+  ngOnChanges(changes: SimpleChanges) {
+    const currentProject$: Observable<Project> = changes.project$.currentValue;
+    if (currentProject$) {
+      currentProject$.subscribe(project => {
+        this.files$ = this.projectService.getProjectFiles(project.title);
+      });
+    }
 
-  openDialog(): void {
-    const dialogRed = this.dialog.open(ProjectSettingsDialog, { data: this.project$ });
   }
 
-  toggleChat(): void {
-    console.log("toggling chat");
-    this.chatService.toggle();
+  setBlobstoreUploadUrl(): void {
+    this.projectService.getBlobstoreUploadUrl().pipe(
+      first()
+    ).subscribe(
+      url => { 
+        this.blobstoreUploadUrl = url;
+        console.log("blobstore upload URL set");
+      },
+      error => console.log("Error getting blobstore upload URL: " + error)
+    );
   }
-}
-
-@Component({
-  selector: 'app-project-settings-dialog',
-  templateUrl: 'project-settings-dialog.html',
-})
-export class ProjectSettingsDialog {
 
-  constructor(
-    public dialogRef: MatDialogRef<ProjectSettingsDialog>,
-    @Inject(MAT_DIALOG_DATA) public project$: Observable<Project>) {}
+  toggleRightDrawer(): void {
```

Comment:
Is there a more descriptive name that RightDrawer here and elsewhere in the code? Is this drawer going to be generic or only hold the file explorer?

Reply:
Thanks for the catch. Actually, all of these references should be removed for now due to the decision to pull chat out of the picture for now, and within the workspace they will be changed to something more like projectInfoDrawer (since project settings are also going into the drawer)